subworkflows/local/shortread_preprocessing.nf

```diff
@@ -0,0 +1,188 @@
+/*
+ * SHORTREAD_PREPROCESSING: Preprocessing and QC for short reads
+ */
+
+include { FASTQC as FASTQC_RAW                                  } from '../../modules/nf-core/fastqc/main'
+include { FASTQC as FASTQC_TRIMMED                              } from '../../modules/nf-core/fastqc/main'
+include { FASTP                                                } from '../../modules/nf-core/fastp/main'
+include { ADAPTERREMOVAL as ADAPTERREMOVAL_PE                   } from '../../modules/nf-core/adapterremoval/main'
+include { ADAPTERREMOVAL as ADAPTERREMOVAL_SE                   } from '../../modules/nf-core/adapterremoval/main'
+include { BOWTIE2_REMOVAL_BUILD as BOWTIE2_HOST_REMOVAL_BUILD   } from '../../modules/local/bowtie2_removal_build'
+include { BOWTIE2_REMOVAL_ALIGN as BOWTIE2_HOST_REMOVAL_ALIGN   } from '../../modules/local/bowtie2_removal_align'
+include { BOWTIE2_REMOVAL_BUILD as BOWTIE2_PHIX_REMOVAL_BUILD   } from '../../modules/local/bowtie2_removal_build'
+include { BOWTIE2_REMOVAL_ALIGN as BOWTIE2_PHIX_REMOVAL_ALIGN   } from '../../modules/local/bowtie2_removal_align'
+include { CAT_FASTQ                                            } from '../../modules/nf-core/cat/fastq/main'
+include { SEQTK_MERGEPE                                        } from '../../modules/nf-core/seqtk/mergepe/main'
+include { BBMAP_BBNORM                                         } from '../..//modules/nf-core/bbmap/bbnorm/main'
+
+workflow SHORTREAD_PREPROCESSING {
+    take:
+    ch_raw_short_reads        // [ [meta] , fastq1, fastq2] (mandatory)
+    ch_host_fasta             // [fasta] (optional)
+    ch_phix_db_file           // [fasta] (optional)
+    ch_metaeuk_db             // [fasta] (optional)
+
+    main:
+    ch_versions = Channel.empty()
+    ch_multiqc_files = Channel.empty()
+    ch_short_reads_assembly = Channel.empty()
+
+    FASTQC_RAW(
+        ch_raw_short_reads
+    )
+    ch_versions = ch_versions.mix(FASTQC_RAW.out.versions.first())
+    ch_multiqc_files = ch_multiqc_files.mix(FASTQC_RAW.out.zip)
+
+    ch_bowtie2_removal_host_multiqc = Channel.empty()
+    if (!params.assembly_input) {
```

Comment:
```if (!params.assembly_input)``` in the future, it feels like we can move away from this. Could the assembly_input logic be determined by the users input?

Reply:
I was wondering about this actually myself:

If you've pre-assembled reads, they should already be cleaned. Which means you could skip this entire subworkflow, so maybe this if else statement could be pushed to the main `mag.nf` around the new subworkflow? @prototaxites what do you think, is that a reasonable assuemption to make (i.e., skip preprocessing of reads)

But what do you mean by 'users input'? This is basically that, if you give the additional extra sheet to `--assembly_input` (that contains the paths to contig FASTAS)  then we can skip assembly. It's not a boolean flag, if that is what you were meaning.

Reply:
> But what do you mean by 'users input'? This is basically that, if you give the additional extra sheet to --assembly_input (that contains the paths to contig FASTAS) then we can skip assembly. It's not a boolean flag, if that is what you were meaning.

Ah, then this makes more sense. I thought it was a boolean.

Reply:
> If you've pre-assembled reads, they should already be cleaned. Which means you could skip this entire subworkflow, so maybe this if else statement could be pushed to the main mag.nf around the new subworkflow?

yes, I actually did this first, but then moved it back into the subworkflow. Not sure what makes the most sense.

Reply:
> If you've pre-assembled reads, they should already be cleaned. Which means you could skip this entire subworkflow, so maybe this if else statement could be pushed to the main mag.nf around the new subworkflow? @prototaxites what do you think, is that a reasonable assuemption to make (i.e., skip preprocessing of reads)

Yes, I think this probably makes the most sense. While I'm sure there's someone out there with the use-case that they want to map a new set of reads against an existing assembly? I think we should assume that the reads accompanying pre-computed assemblies are already clean - someone with that use case probably knows what they are doing!

Personally I'm in favour of putting the `if!(assembly_input)` line in the main WF rather than the subWF; the subWF code should only operate when we actually want to use it. Plus I think it helps make the logic of the main WF clearer by stating when and when not things are run.

Reply:
Thanks for input. I think it makes sense to have it in the main workflow also. If you think this looks fine @jfy133 and @prototaxites I'll go ahead and merge.